### ISSUE TITLE
Missing values and NaN's error

### ISSUE THREAD

**Issue by hoangtt1989** [tinglab/mLDM#2]
I'm attempting to use `mLDM` for OTU count data. My OTU count matrix is of dimension `3122*1249` and my environment factor matrix is of dimension `3122*25`. For the EF matrix, I ensured that the categorical variables were coded as 0 - 1 like in the Colorectal cancer example. There are no missing values in either of my matrices. However, when I attempt to use `mLDM`, I get the following error:

`Error: missing values and NaN's not allowed if 'na.rm' is FALSE`

Do you have any idea why I might be getting this error? I tried reducing the number of OTUs to as few as 70 and I am still getting the error.

**Comment by tinglab** [comment]
Yeah, could you tell me more about when the error arises ? You can set the verbose = TRUE to have a look at the outputs from the mLDM. If the error occurred during the running, I will try to find the reason.

**Comment by hoangtt1989** [comment]
Hi, this is what happens (`OTU` is my count matrix and `sam_des` is my environmental factor matrix):

`> mod <- mLDM(OTU, sam_des, Z_mean = 5, verbose = T)`
`Error: missing values and NaN's not allowed if 'na.rm' is FALSE`

And there are no missing values in either matrix:

`> any(is.na(OTU))`
`[1] FALSE`
`> any(is.na(sam_des))`
`[1] FALSE`

**Comment by tinglab** [comment]
Yeah, at the beginning of the mLDM, the program will calculate the spearman correlation among otus using the OTU table, and the spearman correlation between otus and environmental factors using the otu table and meta table, which can be used in the initialization. One possibe reason may be that some column within your data table has the same values, so that there is the 'NaN' in the correlation matrix. you can compute the correlation via R function 'cor' (e.g. cor(OTU),  cor(OTU, sam_des)), and check it.

**Comment by hoangtt1989** [comment]
I did the following:

`> any(is.nan(cor(OTU)))
[1] FALSE`
`> any(is.nan(cor(OTU, sam_des)))
[1] FALSE`
`> any(is.na(cor(OTU)))
[1] FALSE`
`> any(is.na(cor(OTU, sam_des)))
[1] FALSE`

**Comment by tinglab** [comment]
I'm sorry for your trouble and I add some codes to check the initialization of the mLDM, you can use the mLDM-v1.1.cpp to rerun on your dataset. Surely, if it is possible, you can let me have a look at the test data and help you solve this problem.

**Comment by hoangtt1989** [comment]
Thank you for updating the program. I installed the new version and I'm getting the error

`Initialization Error! NaN found in the spearman correlation matrix among X, or between X and M`

even though I checked the spearman correlation matrices (calculated using `rcorr`) and found no `NaN` or `na` values. I've prepared a subset of my data that I can share with you. How can I send it to you? I would prefer not to attach it here.
